### ISSUE TITLE
Merge `twisted/python/_utilpy3.py` into `twisted/python/util.py`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6235]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6235|
|Type|task|
|Created|2012-12-28 04:03:53Z|
|Branch|https://github.com/twisted/twisted/tree/util-merge-6235|

Also merge the associated test module.  See [#6183](https://github.com/twisted/twisted/issues/6183).

<details><summary>Searchable metadata</summary>

```
trac-id__6235 6235
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_util_merge_6235 branches/util-merge-6235
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1356667433000000 1356667433000000
changetime__1357088723000000 1357088723000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [36657]) Branching to 'util-merge-6235'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

It's done.  I tried to provide good descriptions of the steps and the logic behind them in each commit message.  The result is not as nice as I thought it would be, with respect to change history.  I'm now convinced it's not feasible to restore any more of the history than this branch does (which is to say, any of the history of the moved code); perhaps svnadmin could do it, I don't know, but I don't think I'll try to go down that road.

The code moved into `util.py` and `test_util.py` is all copied verbatim from `_utilpy3.py` and `test_utilpy3.py` except that I had to change `TestCase` to `unittest.TestCase` for the class statements being moved into `test_util.py`.

It may be instructive to consider the diff 36657:36669 separately from r36670 and r36671.  The former is "move stuff around without interestingly changing it" (r36668 not withstanding).  The remainder is "adjust syntax and obliterate unported names" necessary to make the merged modules actually work on Python 3 (only to the level of functionality previously present in `_utilpy3`).

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/util-merge-6235)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince set owner to @tomprince|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to @exarkun|
|-|-|

1. This is missing imports from ```__future__```.
1. ```GetPasswordTest``` but isn't deleted and is skipped due to lack of process support on the reactor. It should be deleted.
1. ```InitGroupsTests```) wasn't ported but is passing. It should be deleted or the appropriate functionality marked as supported. Looking closer, it appears that this isn't intended to be public anyway. (This can be done in a separate ticket).

Please commit after fixing 1+2 (and possibly 3).

(Note for archaeologists: there is no interesting history on trunk in ```util.py``` and ```test_util.py``` that is lost. There is one non-py3 related changeset in ```_utilpy3.py```: r35964 closing [#6063](https://github.com/twisted/twisted/issues/6063)).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [36719]) Merge util-merge-6235

Author: exarkun
Reviewer: tom.prince
Fixes: [#6235](https://github.com/twisted/twisted/issues/6235)

Re-integrate the Python 3 ported miscellaneous util implementations into
`twisted/python/util.py`.
